wordpress-mobile/WordPress-Android
# Create My Site header item

Fixes #13346

This PR adds all the presentation logic to show category headers (layout, ViewHolder) and a SiteItemsBuilder which right now adds only category headers to the list based on the same logic as before. By default there are 4 headers: 
- Publish - always visible
- Configuration - visible if the user has the rights to configure the site - it's either for WP.com admins (`hasCapabilityManageOptions` or `hasCapabilityListUsers` have to be true) or self hosted sites
- Look and feel - visible for admin wp.com users
- External - always visible

<img width="506" alt="Screenshot 2020-11-30 at 11 20 59" src="https://user-images.githubusercontent.com/1079756/100597786-2512c080-32fe-11eb-9dee-c4050fb96b61.png">


To test:
- Turn on the improved my site flag
- Go to My Site
- Notice the headers are visible
- Change your site to a different WP.com site where you're not an administrator
- Notice the "Configuration" is gone
- Notice the "Look and feel" is now gone
- Change your site to a self-hosted site
- Notice the "Configuration" is visible
- Notice the "Look and feel" is still gone

PR submission checklist:
